# Tarea 2

### Autor: Esthefany Mafla

### Librerias

In [2]:
import numpy as np
import pandas as pd

### Datos sobre la felicidad

In [7]:
felicidad=pd.read_csv('https://raw.githubusercontent.com/cienciadedatos/datos-de-miercoles/master/datos/2019/2019-08-07/felicidad.csv'
                    )
felicidad.head(8)

pais  anio  escalera_vida   log_pib  soporte_social  \
0  Afghanistán  2008       3.723590  7.168690        0.450662   
1  Afghanistán  2009       4.401778  7.333790        0.552308   
2  Afghanistán  2010       4.758381  7.386629        0.539075   
3  Afghanistán  2011       3.831719  7.415019        0.521104   
4  Afghanistán  2012       3.782938  7.517126        0.520637   
5  Afghanistán  2013       3.572100  7.522238        0.483552   
6  Afghanistán  2014       3.130896  7.516955        0.525568   
7  Afghanistán  2015       3.982855  7.500539        0.528597   

   expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
0         50.799999  0.718114     0.177889               0.881686   
1         51.200001  0.678896     0.200178               0.850035   
2         51.599998  0.600127     0.134353               0.706766   
3         51.919998  0.495901     0.172137               0.731109   
4         52.240002  0.530935     0.244273               0.775620   
5         52.560001  0.577955     0.070403               0.823204   
6         52.880001  0.508514     0.113184               0.871242   
7         53.200001  0.388928     0.089091               0.880638   

   afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
0         0.517637         0.258195   0.612072           -1.929690   
1         0.583926         0.237092   0.611545           -2.044093   
2         0.618265         0.275324   0.299357           -1.991810   
3         0.611387         0.267175   0.307386           -1.919018   
4         0.710385         0.267919   0.435440           -1.842996   
5         0.620585         0.273328   0.482847           -1.879709   
6         0.531691         0.374861   0.409048           -1.773257   
7         0.553553         0.339276   0.260557           -1.844364   

   calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
0        -1.655084               1.774662                 NaN   
1        -1.635025               1.722688                 NaN   
2        -1.617176               1.878622                 NaN   
3        -1.616221               1.785360                 NaN   
4        -1.404078               1.798283                 NaN   
5        -1.403036               1.223690                 NaN   
6        -1.312503               1.395396                 NaN   
7        -1.291594               2.160618                 NaN   

   gini_banco_mundial_promedio  
0                          NaN  
1                          NaN  
2                          NaN  
3                          NaN  
4                          NaN  
5                          NaN  
6                          NaN  
7                          NaN

In [6]:
felicidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1704 entries, 0 to 1703
Data columns (total 17 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   pais                         1704 non-null   object 
 1   anio                         1704 non-null   int64  
 2   escalera_vida                1704 non-null   float64
 3   log_pib                      1676 non-null   float64
 4   soporte_social               1691 non-null   float64
 5   expectativa_vida             1676 non-null   float64
 6   libertad                     1675 non-null   float64
 7   generosidad                  1622 non-null   float64
 8   percepcion_corrupcion        1608 non-null   float64
 9   afecto_positivo              1685 non-null   float64
 10  afecto_negativo              1691 non-null   float64
 11  confianza                    1530 non-null   float64
 12  calidad_democracia           1558 non-null   float64
 13  calidad_entrega   

### Rango de datos en la varibale año

In [28]:
print("El set de datos sobre felicidad muestra una conjunto de variables tomadas entre los años",  felicidad.anio.min(), "y",felicidad.anio.max())

El set de datos sobre felicidad muestra una conjunto de variables tomadas entre los años 2005 y 2018


### Medidas de tendencia central para el primer y ultimo año de la base de datos.

### 2005

In [116]:
mean2005=felicidad[felicidad['anio']==2005]['escalera_vida'].mean()

std2005=felicidad[felicidad['anio']==2005]['escalera_vida'].std()

In [117]:
print( "Para el año 2005 el promedio de felicidad es de", mean2005, " y una desviacion estandar de", std2005) 

Para el año 2005 el promedio de felicidad es de 6.446164272449635  y una desviacion estandar de 0.9191426322726483


In [53]:
print(felicidad[felicidad['anio']==2005]['escalera_vida'].max())
print(felicidad[felicidad['anio']==2005]['escalera_vida'].min())

8.01893424987793
4.718733787536621


### 2018

In [118]:
mean2018=felicidad[felicidad['anio']==2018]['escalera_vida'].mean()

std2018=felicidad[felicidad['anio']==2018]['escalera_vida'].std()

In [119]:
print( "Para el año 2005 el promedio de felicidad es de", mean2018, " y una desviacion estandar de", std2018) 

Para el año 2005 el promedio de felicidad es de 5.502134340650895  y una desviacion estandar de 1.1034612436939357


In [54]:
print(felicidad[felicidad['anio']==2018]['escalera_vida'].max())
print(felicidad[felicidad['anio']==2018]['escalera_vida'].min())

7.858107089996338
2.694303274154663


De acuerdo con los datos existente se puede decir que el grado de felicidad a nivel global era mayor en el año 2005 que en el año 2018. Si adicional observamos los valore maximos y minimos de la escalera de vida podemos agregar que en el año 2005 la media se econtraba mas centrada que en el año 2018 que presenta una ligera inclinacion hacia el valor máximo. De la misma manera los datos en el año 2018 se encuentran mas dispersos que en el año 2005. Esto puede ser resultado de la inclusion de mas paises en la muestra. como se observa en la siguiente tabla.

In [55]:
felicidad. groupby('anio')['pais'].count()

anio
2005     27
2006     89
2007    102
2008    110
2009    114
2010    124
2011    146
2012    142
2013    137
2014    145
2015    143
2016    142
2017    147
2018    136
Name: pais, dtype: int64

Mientras que en el 2005 se recogen datos para 27 paises en el 2018 el numeo de paises aumenta a 136 que es aproximadamente 4 veces el valor incial. Por esto la comparacion del promedio de felicidad para el primer y ultimo año puede tener sesgo.

### Comparación de valores a nivel mundial con dos países de su elección en el primer año reportado y el último año reportado.

In [107]:
paises2=felicidad[(felicidad['anio']==2005) | (felicidad['anio']==2018)].sort_values(by='anio').groupby('pais')['pais'].count()==2

In [109]:
paises2[paises2==True]

pais
Alemania           True
Arabia Saudita     True
Australia          True
Brasil             True
Bélgica            True
Canadá             True
Dinamarca          True
Egipto             True
España             True
Francia            True
Greece             True
Iran               True
Italia             True
Japón              True
Jordania           True
Líbano             True
México             True
Nigeria            True
Pakistan           True
Países Bajos       True
Reino Unido        True
República Checa    True
Rumania            True
Suecia             True
Turquía            True
Venezuela          True
Name: pais, dtype: bool

Se eligen los paises de España y Mexico.

### Acceso a la información de la felicidad del primer y último año para los dos países

### España 2005 y 2018


In [133]:
españa=felicidad[(felicidad['anio']==2005) | (felicidad['anio']==2018)]
españa=españa[españa['pais']=='España']
españa

pais  anio  escalera_vida    log_pib  soporte_social  \
1399  España  2005       7.152786  10.401288        0.961043   
1411  España  2018       6.513371  10.465594        0.910315   

      expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
1399         71.500000  0.916165          NaN               0.777272   
1411         74.400002  0.722251    -0.079351               0.776504   

      afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
1399         0.775784         0.240643   0.527846            0.663309   
1411         0.659188         0.357191   0.285196                 NaN   

      calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
1399         1.319199               1.724895               0.324   
1411              NaN               1.667077                 NaN   

      gini_banco_mundial_promedio  
1399                     0.345385  
1411                     0.345385

### Mexico 2005 y 2018

In [135]:
mexico=felicidad[(felicidad['anio']==2005) | (felicidad['anio']==2018)]
mexico=mexico[mexico['pais']=='México']
mexico

pais  anio  escalera_vida   log_pib  soporte_social  expectativa_vida  \
978  México  2005       6.580658  9.667230        0.902808         66.199997   
990  México  2018       6.549579  9.769919        0.858069         68.300003   

     libertad  generosidad  percepcion_corrupcion  afecto_positivo  \
978  0.813745          NaN               0.764249         0.819803   
990  0.816200    -0.181818               0.808638         0.881713   

     afecto_negativo  confianza  calidad_democracia  calidad_entrega  \
978         0.218943   0.426359           -0.124936        -0.088847   
990         0.213254   0.294568                 NaN              NaN   

     de_escalera_pais_anio  gini_banco_mundial  gini_banco_mundial_promedio  
978               2.235451               0.489                       0.4698  
990               2.326330                 NaN                       0.4698

In [146]:
mexico[mexico['anio']==2005]['escalera_vida'].values

array([6.58065796])

### Comparacion

In [154]:
comp=pd.DataFrame({'año':[2005,2018],
                   'EC General':[mean2005, mean2018],
                   'EV España':[españa[españa['anio']==2005]['escalera_vida'].values[0], 
                                españa[españa['anio']==2018]['escalera_vida'].values[0]], 
                   'EV Mexico':[mexico[mexico['anio']==2005]['escalera_vida'].values[0], 
                                mexico[mexico['anio']==2018]['escalera_vida'].values[0]]})
comp

año  EC General  EV España  EV Mexico
0  2005    6.446164   7.152786   6.580658
1  2018    5.502134   6.513371   6.549579

### Interpretacion

* De acuerdo con la tabla resumen se puede evidenciar que en el año 2005 tanto Mexico como España se encontraban sobre el nivel de felicidad global, no obstante España tiene un grado de felicidad superior al de Mexico. 

* En el año 2018 el comportamiento es similar ya que España y Mexico mantiene niveles de felicidad sobre el promedio global, en este caso se puede notar que el grado de felicidad de España con respecto a Mexico es ligeramente superior.

* Por lo mencionado anteriormente podemos decir que a nivel global el grado de felicidad dismunuye entre el 2005 y el 2018 este comportamiento es heredado para el pais de España que presenta una disminision del grado de felicidad mucho mas marcado que la disminucion que se da en Mexico. 